src/spacerini/index/index.py

```diff
@@ -157,25 +163,47 @@ def index_streaming_hf_dataset(
         Name of HuggingFace dataset to stream
     split : str
         Split of dataset to index
-    column_to_index : str
+    column_to_index : List[str]
         Column of dataset to index
+    doc_id_column : str
+        Column of dataset to use as document ID
     ds_config_name: str
         Dataset configuration to stream. Usually a language name or code
-    num_rows : str
+    num_rows : int
         Number of rows in dataset
-    disable_tqdm : bool
-        Disable tqdm output
     
-    See io.anserini.IndexCollection.Args for remaining argument definitions
-    https://github.com/castorini/anserini
+    See [docs](../../docs/arguments.md) for remaining argument definitions
 
     Returns
     -------
     None
     """
+    
     args = parse_args(**locals(), for_otf_indexing=True)
-    ds = load_dataset(ds_path, name=ds_config_name, split=split, streaming=True)
+    if os.path.exists(ds_path):
+        ds = load_from_local(ds_path, split=split, streaming=True)
+    else:
+        ds = load_from_hub(ds_path, split=split,config_name=ds_config_name, streaming=True)
+
     indexer = LuceneIndexer(args=args)
+
     for i, row in tqdm(enumerate(ds), total=num_rows, disable=disable_tqdm):
-        indexer.add(json.dumps({"id": i, "contents": row[column_to_index]}))
+        contents = " ".join([row[column] for column in column_to_index])
+        indexer.add(json.dumps({"id": i if not doc_id_column else row[doc_id_column] , "contents": contents}))
```

Comment:
Does JsonCollection limit us to only id and content fields? I assume that's why you're concatenating in #191

Reply:
Yeah It does